src/plugins/data/common/search/search_source/search_source.ts

```diff
@@ -539,7 +538,7 @@ export class SearchSource {
       if (!body.hasOwnProperty('_source')) {
         body._source = sourceFilters;
       }
-      if (body._source.excludes) {
+      if (body._source && body._source.excludes) {
```

Comment:
I'm not sure if the check for `body._source` is needed? `indexPatterns.getSourceFiltering()` will always return something, even if it's just `{ excludes: [] }`, so if `_source` isn't previously set it will already be set in L539.